server/pages/index.js

```diff
@@ -119,7 +122,19 @@ function getCurrentCommitShortChecksum() {
 	}
 }
 
-function getDefaultContext( request ) {
+function getFileContentsSafely( filepath ) {
+	return new Promise( function( resolve ) {
+		fs.readFile( filepath, function( err, content ) {
```

Comment:
@ehg @DanReyLop 

I've pushed up my caching implementation. It's up for discussion/debate! 😄 

I opted to create two Promises at boot time, one for each file to be inlined. The Promises will resolve once the file contents are read from the disk.

Later, in `getDefaultContext`, I use `Promise.all` with those two Promises. The idea is that the file reading happens async, and it is initiated at boot time. At runtime, chances are the files have already been loaded, so the request will not block. It is possible that the first request comes in before the Promises resolve, but I'm thinking it's pretty unlikely.

@ehg After our Slack discussion on the issues with Promises, I'm not sure whether we need to worry about them in this case. I don't want to be overly optimistic though! Let's think about it, and discuss further.